src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -10,18 +10,91 @@
 constexpr auto DEFAULT_QUEUE_PATH = "queue/sockets/queue";
 constexpr auto REPORTS_QUEUE_PATH = "queue/vd/reports";
 constexpr auto REPORTS_BULK_SIZE {1};
 constexpr auto REPORTS_THREADS_NUMBER {1};
 constexpr auto MICROSEC_FACTOR {1000000};
+constexpr auto VD_STATE_QUEUE_PATH = "queue/vd/state_track";
+constexpr auto VD_KEYSTORE_PATH = "queue/keystore";
+constexpr auto VD_DATABASE_PATH {"queue/vd"};
+
+bool VulnerabilityScannerFacade::decompressDatabase(const std::string& databaseVersion)
+{
+    bool ret = false;
+
+    // Check database version and presence. It will attempt to decompress the database
+    // if the version doesn't not match, the state_track does not have information
+    // or there's no VD DB.
+    if (databaseVersion.compare(__ossec_version) != 0 || databaseVersion.empty() ||
+        !std::filesystem::exists(VD_DATABASE_PATH))
+    {
+        // Search in path for a compressed database file.
+        std::regex regexPattern(VD_COMPRESSED_FILE_PATTERN);
+        std::string compressedFilePath;
+        for (const auto& item : std::filesystem::directory_iterator("./"))
+        {
+            if (std::regex_match(item.path().filename().string(), regexPattern))
+            {
+                compressedFilePath = item.path().filename().string();
+                break;
+            }
+        }
+
+        // Check for XZ downloaded file .
+        if (!compressedFilePath.empty())
+        {
+            logInfo(WM_VULNSCAN_LOGTAG, "Starting database file decompression.");
+            logDebug2(WM_VULNSCAN_LOGTAG, "Starting XZ file decompression");
+
+            std::string decompressedFilePath = compressedFilePath;
+            Utils::replaceFirst(decompressedFilePath, ".xz", "");
+
+            // Decompress XF file format.
+            Utils::XzHelper(std::filesystem::path(compressedFilePath), std::filesystem::path(decompressedFilePath))
+                .decompress();
+
+            // Extract queue/vd and queue/vd_updater
+            std::vector<std::string> extractOnly;
+            extractOnly.emplace_back("queue/vd");
+
+            // Decompress also keystore if missing.
+            if (!std::filesystem::exists(VD_KEYSTORE_PATH))
+            {
+                extractOnly.emplace_back(VD_KEYSTORE_PATH);
+            }
+
+            logDebug2(WM_VULNSCAN_LOGTAG, "Starting TAR file decompression");
+
+            // Decompress TAR file format.
+            Utils::ArchiveHelper::decompress(decompressedFilePath, "", extractOnly);
+
+            // Cleanup
+            std::filesystem::remove_all(compressedFilePath);
+            std::filesystem::remove_all(decompressedFilePath);
+
+            ret = true;
+
+            logInfo(WM_VULNSCAN_LOGTAG, "Database decompression finished.");
+        }
+        else
+        {
+            throw std::runtime_error("Missing database compressed file. Check DOWNLOAD_CONTENT option.");
+        }
```

Comment:
Nitpick: IMO is better to prevent indentation when the code inside the if is too long and the else terminates the function
```suggestion
    if (compressedFilePath.empty())
    {
        throw std::runtime_error("Missing database compressed file. Check DOWNLOAD_CONTENT option.");
    }
    
    logInfo(WM_VULNSCAN_LOGTAG, "Starting database file decompression.");
    logDebug2(WM_VULNSCAN_LOGTAG, "Starting XZ file decompression");

    std::string decompressedFilePath = compressedFilePath;
    Utils::replaceFirst(decompressedFilePath, ".xz", "");

    // Decompress XF file format.
    Utils::XzHelper(std::filesystem::path(compressedFilePath), std::filesystem::path(decompressedFilePath))
        .decompress();

    // Extract queue/vd and queue/vd_updater
    std::vector<std::string> extractOnly;
    extractOnly.emplace_back("queue/vd");

    // Decompress also keystore if missing.
    if (!std::filesystem::exists(VD_KEYSTORE_PATH))
    {
        extractOnly.emplace_back(VD_KEYSTORE_PATH);
    }

    logDebug2(WM_VULNSCAN_LOGTAG, "Starting TAR file decompression");

    // Decompress TAR file format.
    Utils::ArchiveHelper::decompress(decompressedFilePath, "", extractOnly);

    // Cleanup
    std::filesystem::remove_all(compressedFilePath);
    std::filesystem::remove_all(decompressedFilePath);

    ret = true;

    logInfo(WM_VULNSCAN_LOGTAG, "Database decompression finished.");
```

Reply:
I like it